test/document/document_test.ts

```diff
@@ -638,6 +638,34 @@ describe('Document', function () {
     }
   });
 
+  it('text garbage collection test', function () {
+    const doc = Document.create<{ text: PlainText }>('test-col', 'test-doc');
+    doc.update((root) => root.createText('text'));
+    doc.update((root) => root.text.edit(0, 0, 'ABCD'));
+    doc.update((root) => root.text.edit(0, 2, '12'));
+
+    assert.equal(
+      '[0:00:0:0 ][3:00:1:0 12]{2:00:1:0 AB}[2:00:1:2 CD]',
+      doc.getRoot().text.getAnnotatedString(),
+    );
+
+    assert.equal(1, doc.getGarbageLen());
+    doc.garbageCollect(MaxTimeTicket);
+    assert.equal(0, doc.getGarbageLen());
+
+    assert.equal(
+      '[0:00:0:0 ][3:00:1:0 12][2:00:1:2 CD]',
+      doc.getRoot().text.getAnnotatedString(),
+    );
+
+    doc.update((root) => root.text.edit(2, 4, ''));
+
+    assert.equal(
+      '[0:00:0:0 ][3:00:1:0 12]{2:00:1:2 CD}',
+      doc.getRoot().text.getAnnotatedString(),
+    );
+  });
+
```

Comment:
I'm curious if the below is not able to check the bug 🤔

https://github.com/yorkie-team/yorkie-js-sdk/blob/93548006e50de7338ea372c0032ce868fcbe130e/test/document/document_test.ts#L383-L414

Reply:
`insPrev` is used to refer to the previous node inserted together but split by `node.split`. The problem occurs only when the `insPrev` node is deleted then the target node is referenced.

```typescript
    doc.update((root) => root.text.edit(0, 0, 'ABCD'));
    doc.update((root) => root.text.edit(0, 2, '12'));

    // 'CD' points to 'AB' as `insPrev`
    assert.equal(
      '[0:00:0:0 ][3:00:1:0 12]{2:00:1:0 AB}[2:00:1:2 CD]',
      doc.getRoot().text.getAnnotatedString(),
    );

    assert.equal(1, doc.getGarbageLen());
    doc.garbageCollect(MaxTimeTicket);
    assert.equal(0, doc.getGarbageLen());

    // But `AB` has been removed by GC.
    assert.equal(
      '[0:00:0:0 ][3:00:1:0 12][2:00:1:2 CD]',
      doc.getRoot().text.getAnnotatedString(),
    );

    // When trying to remove the `CD`, a script error occurred because the `insPrev` `AB` was deleted.
    doc.update((root) => root.text.edit(2, 4, ''));

    assert.equal(
      '[0:00:0:0 ][3:00:1:0 12]{2:00:1:2 CD}',
      doc.getRoot().text.getAnnotatedString(),
    );
```